modules/xmpp/JingleSessionPC.js

```diff
@@ -962,98 +963,106 @@ JingleSessionPC.prototype.generateNewStreamSSRCInfo = function () {
 };
 
 /**
- * Remove streams.
- * @param stream stream that will be removed.
- * @param callback callback executed after successful stream addition.
- * @param errorCallback callback executed if stream addition fail.
- * @param ssrcInfo object with information about the SSRCs associated with the
- * stream.
+ * Remove stream handling for firefox
+ * @param stream: webrtc media stream
  */
-JingleSessionPC.prototype.removeStream = function (stream, callback, errorCallback,
-    ssrcInfo) {
-    // Conference is not active
-    if(!this.peerconnection) {
-        callback();
+JingleSessionPC.prototype._handleFirefoxRemoveStream = function (stream) {
+    if (!stream) { //There is nothing to be changed
+        return;
+    }
+    var sender = null;
+    // On Firefox we don't replace MediaStreams as this messes up the
+    // m-lines (which can't be removed in Plan Unified) and brings a lot
+    // of complications. Instead, we use the RTPSender and remove just
+    // the track.
+    var track = null;
+    if (stream.getAudioTracks() && stream.getAudioTracks().length) {
+        track = stream.getAudioTracks()[0];
+    } else if (stream.getVideoTracks() && stream.getVideoTracks().length) {
+        track = stream.getVideoTracks()[0];
+    }
+
+    if (!track) {
+        var msg = "Cannot remove tracks: no tracks.";
+        logger.log(msg);
         return;
     }
 
-    // Remember SDP to figure out added/removed SSRCs
-    var oldSdp = null;
+    // Find the right sender (for audio or video)
+    this.peerconnection.peerconnection.getSenders().some(function (s) {
+        if (s.track === track) {
+            sender = s;
+            return true;
+        }
+    });
 
-    if(this.peerconnection.localDescription) {
-        oldSdp = new SDP(this.peerconnection.localDescription.sdp);
+    if (sender) {
+        this.peerconnection.peerconnection.removeTrack(sender);
+    } else {
+        logger.log("Cannot remove tracks: no RTPSender.");
     }
+};
 
-    if(!oldSdp) {
-        callback();
+/**
+ * Just remove the stream from the peerconnection
+ * @param stream either the low-level webrtc MediaStream or 
+ *  a Jitsi mediastream
+ * NOTE: must be called within a work function being executed
+ *  by the modification queue.
+ */
+JingleSessionPC.prototype.removeStreamFromPeerConnection = function (stream, stopStream) {
+    let actualStream = stream && stream.getOriginalStream ? stream.getOriginalStream() : stream;
+    if (!this.peerconnection) {
         return;
     }
-
     if (RTCBrowserType.getBrowserType() ===
```

Comment:
yeah i think that sounds like a better take on it, if you don't think it's too bad for now though i agree it'd be nice to have that in a separate PR